ariadne/src/Ariadne/Wallet/Backend/KeyStorage.hs

```diff
@@ -172,6 +173,22 @@ resolveWalletRef walletSelRef walRef db = case walRef of
     walletList :: [HdRoot]
     walletList = toList (readAllHdRoots hdWallets)
 
+-- | Like 'resolveWalletRef', but also reads some data from 'DB'
+-- corresponding to the resolved root ID, assuming that the wallet
+-- with this ID definitely exists.
+resolveWalletRefThenRead
```

Comment:
No, this function is very useful. When we resolve wallet ref, we only get a `HdRootId`, but we most certainly want to do something with this id, probably read something from DB corresponding to this id.
Currently most of the code relies on the fact that `resolveWalletRef` returns id of a wallet that definitely exists in DB (where it's relevant). Which is more or less true, but can be wrong if concurrent modifications occur. AFAIU currently they can't occur and we also have AD-274. Anyway, if such inconsistency occurs we treat it as a programmer mistake.
Sadly, it's not easy to represent this information at the type level. This function does it to some extent, so that its caller doesn't need to handle a programmer mistake case.
Now, regarding usages. Yes, currently it's only used in `Tx.hs`, but that's merely because this PR is about `Tx.hs`, so I didn't bother to find other places where `error` calls can be eliminated using this function. In fact, I wanted to created a separate issue to eliminate some `error`s introduced by AD-135, but apparently I haven't done it yet.
Quick search showed me the following:
1. First usage of `resolveWalletRef` is in `resolveAccountRef` and there we read accounts for this wallet. If it fails, we throw an exception in `IO`. Which is more or less OK, but according to CSL guidelines which we don't explicitly follow (yet), but which are still good, programmer mistakes should be handled using `error`, not `IO` exceptions.
2. Second usage is in `newAccount`. There we have this:
```
        -- TODO: make it global
    -- The same function as in `select`. It is unsafe -> not global.
    -- It is assumed that hdRootId is a valid one.
    -- It is, because resolveWalletRef always returns a valid HdRootId.
    getAccounts :: HdWallets -> HdRootId -> IxSet HdAccount
    getAccounts wallets rootId = fromRight
      (error "Bug: UnknownHdRoot")
      (readAccountsByRootId rootId wallets)
```
My function allows to avoid this `error` call and deprecates this comment.
3. Next usage is in `select`. There we have the same `getAccounts`.

I didn't search further, because I suppose that's convincing enough already.
During more profound refactoring we should consider adding more such helpers (e. g. for `resolveAccountRef`), but I didn't think about it yet.

Reply:
This sounds quite convincing, thank you.

Reply:
https://issues.serokell.io/issue/AD-302